File: patch_cop_loading.js
const fs = require('fs');
let file = fs.readFileSync('pages/CommunityOfPractices.tsx', 'utf8');

const target = `const unsubscribe = onSnapshot(q, (snapshot) => {
            const data: CopEvent[] = [];
            snapshot.forEach(doc => data.push(doc.data() as CopEvent));
            setEvents(data);
            setLoading(false);
        });`;

const replacement = `const unsubscribe = onSnapshot(q, (snapshot) => {
            const data: CopEvent[] = [];
            snapshot.forEach(doc => data.push(doc.data() as CopEvent));
            setEvents(data);
            setLoading(false);
        }, (error) => {
            console.error("CoP fetch error:", error);
            showToast('Gagal memuat data: ' + error.message, 'error');
            setLoading(false);
        });`;

file = file.replace(target, replacement);
fs.writeFileSync('pages/CommunityOfPractices.tsx', file);

File: patch_cop_loading.py
import os

with open('pages/CommunityOfPractices.tsx', 'r') as f:
    content = f.read()

target = """        const unsubscribe = onSnapshot(q, (snapshot) => {
            const data: CopEvent[] = [];
            snapshot.forEach(doc => data.push(doc.data() as CopEvent));
            setEvents(data);
            setLoading(false);
        });"""

replacement = """        const unsubscribe = onSnapshot(q, (snapshot) => {
            const data: CopEvent[] = [];
            snapshot.forEach(doc => data.push(doc.data() as CopEvent));
            setEvents(data);
            setLoading(false);
        }, (error) => {
            console.error("CoP fetch error:", error);
            setLoading(false);
        });"""

content = content.replace(target, replacement)

with open('pages/CommunityOfPractices.tsx', 'w') as f:
    f.write(content)
